consistency/tool/clustering/clustering.py

```diff
@@ -43,16 +44,74 @@ def extract_indices_dice(x, y):
         data = np.arange(len(self.words)).reshape(-1, 1)
     
         try:
-            db = dbscan(data, metric=extract_indices_dice, eps=0.3, min_samples=2, algorithm='brute')
+            _, labels = dbscan(data, metric=extract_indices_dice, eps=0.3, min_samples=2, algorithm='brute')
         except Exception as e:
             print(e, " Word Clustering Failed!")
             return None
     
         #if all labels are -1, DBSCAN detected no possible clusters. 
-        if np.count_nonzero(db[1] == -1) == db[1].size:
+        if np.count_nonzero(labels == -1) == labels.size:
             warnings.warn("There were no clusters detected. All words are unique.")
     
-        return db[1]
+        return labels
+
+    def create_word_groups(self):
+        
+
+        labels = np.array(self.cluster())
+
+        temp_dict  = {}
+        if not labels.size or not self.words:
+            return None
+        for j in range(len(self.words)):
+            word_label = labels[j]
+    
+            if word_label in temp_dict:
+                temp_dict[word_label].append(self.words[j])
+                        
+            else:
+                temp_dict[word_label] = [self.words[j]]
+
+        word_groups = []
+
+        def find_clusterID(similar_list):
+            if len(similar_list) == 0 or similar_list == None:
+                return None
+            counts = Counter(similar_list)
+            max_count = counts.most_common(1)[0][1]
+            most_freq_words = [value for value, count in counts.most_common() if count == max_count]
+            most_freq_words.sort() 
+
+            return most_freq_words[0]  
+
+        for k in temp_dict.keys():
+            word_group = wg.WordGroup()
+            if k == -1:
+                word_group.id = "NOISE_WORDS"
+                word_group.kind = "WordGroup"
+                map = dict(Counter(temp_dict[k]))
+                for k, v in map.items():
+                    word_group.word_frequency[k] = v
+                word_groups.append(word_group)
+            else:
+                similar_words = temp_dict[k]
+                id = find_clusterID(similar_words)
+
+                if id is None:
+                    continue
+                word_group.id = id
+                word_group.kind = "WordGroup"
+
+                map = dict(Counter(temp_dict[k]))
+                for k, v in map.items():
+                    word_group.word_frequency[k] = v
+
+                word_groups.append(word_group)
```

Comment:
nit: Line 103-107 are common in both if and else block. It can be moved outside to make it more compact.